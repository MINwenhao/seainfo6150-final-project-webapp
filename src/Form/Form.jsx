import React, { useEffect, useState } from "react";
import { Link } from "react-router-dom";

import styles from "./Form.module.css";

const Form = (props) => {
  // const submittedForm = props.submittedForm;

  const setName = props.setName;
  const [submittedForm, setSubmittedForm] = useState()
  function onSubmit(e) {
    e.preventDefault();
    const data = new FormData(e.target);
    setSubmittedForm(data);
  };
  useEffect(() => {
    if (submittedForm) {
      setName({
        firstName: submittedForm.get("firstName")
        , lastName: submittedForm.get("lastName")
      })

    }
  }, [submittedForm, setName])



  return (
    <div>
      {
        submittedForm ? (
          <Link to="/" >Thanks for your Sign up,Please Click Here Return to Home Page</Link>
        ) : (

            <form onSubmit={onSubmit} className={styles.container}>
              <input type="text" id="first" name="firstName" placeholder="First Name" required />
              <input type="text" id="last" name="lastName" placeholder="Last Name" required />
              <input type="password" id="password" name="password" placeholder="Password" required />
              <input type="email" name="email" id="email" placeholder="Email" required />
              <div>
                <label htmlFor="Gender">Gender</label>
                <select name="Gender" id="Gender">
                  <option value="Male">Male</option>
                  <option value="Female">Female</option>
                </select>
              </div>
              <div>
                <label htmlFor="position">Position</label>
                <select name="position" id="position">
                  <option value="Student">Student</option>
                  <option value="designer">designer</option>
                  <option value="programmer">programmer</option>
                  <option value="others">others</option>
                </select>
              </div>
              <p> select your age range</p>
              <div>
                <input type="radio" name="ageRange" id="adolescent" />
                <label htmlFor="adolescent">less than 19</label>
              </div>
              <div>
                <input type="radio" name="ageRange" id="youngAdult" />
                <label htmlFor="youngAdult">19-30</label>
              </div>
              <div>
                <input type="radio" name="ageRange" id="middleAge" />
                <label htmlFor="middleAge">31-60</label>
              </div>
              <div>
                <input type="radio" name="ageRange" id="oldAdult" />
                <label htmlFor="oldAdult">more than 60</label>
              </div>
              <div>
                <input type="submit" value="confirm" />
              </div>
            </form>
          )
      }
    </div>
  )
}

export default Form
