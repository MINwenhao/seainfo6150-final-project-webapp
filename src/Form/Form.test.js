import { render } from "@testing-library/react";
import React from "react";
import Form from "./Form.jsx";

describe("AddingMachineHooks tests", () => {
  it("renders correctly", () => {
    const { container } = render(<Form />);
    expect(container).toMatchSnapshot();
  });
});