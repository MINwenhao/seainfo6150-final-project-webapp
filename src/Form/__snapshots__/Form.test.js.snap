// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddingMachineHooks tests renders correctly 1`] = `
<div>
  <div>
    <form
      class="container"
    >
      <input
        id="first"
        name="firstName"
        placeholder="First Name"
        required=""
        type="text"
      />
      <input
        id="last"
        name="lastName"
        placeholder="Last Name"
        required=""
        type="text"
      />
      <input
        id="password"
        name="password"
        placeholder="Password"
        required=""
        type="password"
      />
      <input
        id="email"
        name="email"
        placeholder="Email"
        required=""
        type="email"
      />
      <div>
        <label
          for="Gender"
        >
          Gender
        </label>
        <select
          id="Gender"
          name="Gender"
        >
          <option
            value="Male"
          >
            Male
          </option>
          <option
            value="Female"
          >
            Female
          </option>
        </select>
      </div>
      <div>
        <label
          for="position"
        >
          Position
        </label>
        <select
          id="position"
          name="position"
        >
          <option
            value="Student"
          >
            Student
          </option>
          <option
            value="designer"
          >
            designer
          </option>
          <option
            value="programmer"
          >
            programmer
          </option>
          <option
            value="others"
          >
            others
          </option>
        </select>
      </div>
      <div
        class="agePart"
      >
        <p>
           select your age range
        </p>
        <div>
          <input
            id="adolescent"
            name="ageRange"
            type="radio"
          />
          <label
            for="adolescent"
          >
            less than 19
          </label>
        </div>
        <div>
          <input
            id="youngAdult"
            name="ageRange"
            type="radio"
          />
          <label
            for="youngAdult"
          >
            19-30
          </label>
        </div>
        <div>
          <input
            id="middleAge"
            name="ageRange"
            type="radio"
          />
          <label
            for="middleAge"
          >
            31-60
          </label>
        </div>
        <div>
          <input
            id="oldAdult"
            name="ageRange"
            type="radio"
          />
          <label
            for="oldAdult"
          >
            more than 60
          </label>
        </div>
      </div>
      <div>
        <input
          class="button"
          type="submit"
          value="confirm"
        />
      </div>
    </form>
  </div>
</div>
`;
